plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -862,4 +865,92 @@ public function disable( $args, $assoc_args ) {
 		}
 	}
 
+	/**
+	 * When HPOS is enabled, this command lets you remove redundant data from the postmeta table for migrated orders.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <all|id|range>...
+	 * : ID or range of orders to clean up.
+	 *
+	 * [--batch-size=<batch-size>]
+	 * : Number of orders to process per batch. Applies only to cleaning up of 'all' orders.
+	 * ---
+	 * default: 500
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *    # Cleanup post data for order 314.
+	 *    $ wp wc hpos cleanup 314
+	 *
+	 *    # Cleanup postmeta for orders with IDs betweeen 10 and 100 and order 314.
+	 *    $ wp wc hpos cleanup 10-100 314
+	 *
+	 *    # Cleanup postmeta for all orders.
+	 *    wp wc hpos cleanup all
+	 *
+	 *    # Cleanup postmeta for all orders with a batch size of 200 (instead of the default 500).
+	 *    wp wc hpos cleanup all --batch-size=200
+	 *
+	 * @param array $args       Positional arguments passed to the command.
+	 * @param array $assoc_args Associative arguments (options) passed to the command.
+	 * @return void
+	 */
+	public function cleanup_post_data( array $args = array(), array $assoc_args = array() ) {
+		if ( ! $this->synchronizer->custom_orders_table_is_authoritative() || $this->synchronizer->data_sync_is_enabled() ) {
```

Comment:
(This is a half-finished thought, but I'll add it anyway.)

I wondered if it would increase robustness to move this check/duplicate it so it also runs inside the `foreach` loop further down—on the basis that if these conditions change while cleanup is in progress, we might wish to abort.

Of course, there are challenges with that because the value of the relevant options containing this information will have been cached at the start of the (CLI) request. We could overcome that with invalidation, but anyway this made me wonder if perhaps we should set some sort of flag (clean-up-in-progress) that can be used from the settings UI to discourage any changes.

Overkill?